talkative_robot_2.rb

```diff
@@ -0,0 +1,307 @@
+require 'pry'
+
+class Person
+  
+  attr_accessor :name, :gender, :age, :grocery_list
+  
+  def initialize(args)
+    @name   			= args[:name]   || nil
+    @gender 			= args[:gender] || nil
+    @age    			= args[:age]    || 0
+    @grocery_list = nil
+  end
+
+  def self.get_name
+  puts "Hi, what's your name?"
+  @name = gets.chomp.capitalize
+  return @name
+  end
+
+  def self.get_gender
+    puts "Are you male or female? [M/F]"
+    gender = gets.chomp.capitalize
+    until gender == "M" || gender == "F"
+      puts "Are you male or female? [M/F]"
+      gender = gets.chomp.capitalize
+    end
+    case 
+    when gender == "M"
+      @gender = "male"
+    when gender == "F"
+      @gender = "female"  
+    end
+    return @gender
+  end 
+
+  def self.get_age
+    puts "What's your age?"
+    @age = gets.chomp.to_i
+    until @age > 0
+      puts "Please enter a positive number for your age:"
+      @age = gets.chomp.to_i
+    end
+    return @age
+  end 
+
+
+  def self.get_grocery_list
+    print "What items are on your grocery list? "
+    puts "[Please enter all items separated by a comma, then press <enter>]"
+    @grocery_list = gets.chomp.split(", ")
+    return @grocery_list
+  end
+  	# user_grocery_list = GroceryList.new("user_grocery_list.txt", @name)
+
+    # user_grocery_list = gets.chomp.split(", ") 
+    # return user_grocery_list
+    # user_grocery_list = IO.write("user_grocery_list.txt", user[:name])
+    # # user_grocery_list = GroceryList.New("grocery_list.txt", user[:name])
+  
+
+  # def to_s
+  #   "#{@name} is a #{@age}-year-old #{@gender}."
+  # end
+end 
+
+class User < Person
+  
+  attr_accessor :name, :gender, :age, :terror
+
+  def initialize(args)
+    super
+    @terror            = 0
+ 
+  end
+
+  def robot_terror_scale
+    puts "On a scale of 1 - 10, with 1 being the least, how afraid of robots are you?"
+    @terror = gets.chomp.to_i
+    puts "You should be at more like a #{terror * 10}. I moonlight as an assassin."
+    @terror
+  end 
+
+  # def to_s
+  #   "#{@name} is a #{@age}-year-old #{@gender} who fears robots at #{@terror}/10, 
+  #    which isn't nearly enough."
+  # end
+
+  def male?(gender)
```

Comment:
Is this specific to `User`s and not to all `Person`s? Also, you don't need to pass in any arguments now that we have instance variables.